R/integer_vector.R

```diff
@@ -13,6 +13,14 @@
 #'                   is at most the specified length, inclusive
 #'                   Note: if exact_length is specified, this parameter is
 #'                   ignored
+#' @param min the minimum allowed value for each vector element,
```

Comment:
by convention, when we are talking about vectors, we use _value/_values postfix for arguments that refer to the values inside the vectors. This is the reason why allow_na for is_real_value becomes allow_na_values for is_real_vector. 

To stick to the same convention, we should have min_value and max_value and inclusive_min_value, and inclusive_max_value. 

I am not completely happy to be honest, but the problem is that allow_na was pulling double duty in the vector check routines: if it was allow_na, did it mean it only accepted one single NA instead of the vector, or that it accepted NA anywhere in the vector?